### PR TITLE
Update vehicle configuration parameters for CDAR VRU UC testing

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
Cherry Picking shared from https://github.com/usdot-fhwa-stol/carma-config/tree/spike-smoke-test branch. This change set is meant to be related to Carma Platform configurations.

**Notable differences to smoke testing branch**
- Removed duplicate VehicleConfigParams.yaml in `ail_vru_uc1_scenario/cdasim_config/carma`. 

## Motivation and Context
CDAR-593

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by MishkaMN** [comment]
Please add back the VehicleConfigParams.yml file. That is not a duplicate file, but meant to provide easy way to modify the parameters in there for the tester just like all other files

**Review by MishkaMN — CHANGES_REQUESTED** [review]
And also the docker-compose.yml docker volume to it

File: ail_vru_uc1_scenario/carma_rosconsole.conf [hunk 1]
```diff
@@ -24,3 +24,6 @@ log4j.logger.ros.route_following_plugin=INFO
 log4j.logger.ros.pure_pursuit_wrapper=INFO
 log4j.logger.ros.system_controller=INFO
 log4j.logger.ros.subsystem_controllers=INFO
+log4j.logger.ros.lci_strategic_plugin=DEBUG
+log4j.logger.ros.stop_and_wait_plugin=DEBUG
+log4j.logger.ros.yield_plugin=DEBUG
```
Comment (adamlm) [inline ail_vru_uc1_scenario/carma_rosconsole.conf:29]: This was added during debugging, so it can be safely removed.

**Review by adamlm — CHANGES_REQUESTED** [review]

**Comment by paulbourelly999** [comment]
> Please add back the VehicleConfigParams.yml file. That is not a duplicate file, but meant to provide easy way to modify the parameters in there for the tester just like all other files

I would strongly advise against having a separate VehcileConfigParam file just to avoid having to rebuild carma-config images in our release. This has multiple draw backs:
- Confusing ( Which one is being used, what if they are different)
- Breaking Convention ( Every CARMA Config provides a VehicleConfigParam.yaml file that is built into the image volume, doing something different here may encourage other down the line to also diverge from this convention)
- Duplication ( Requires more maintenance and is more error prone. You make a change in one file and forget to push it to the other file.)

Please let me know  if I am missing something here.

**Comment by MishkaMN** [comment]
My reasoning to have it in the config folder is because using carma-config image itself has drawbacks. 
- Just changing one parameter is not user friendly at all -> change file + build image + set image just to exactly mirror what I have locally. I'd say let's just change the file and be done with it.
- More importantly, I imagine it is also error prone because the tester may forget to do those extra steps because all other parameter changes only require local file changes. So they may end up doing many runs until after the fact. This error is hard to notice because the actual value being used in the system is hidden from the user although local file shows updated value. We have an opportunity to lessen that room for error for the tester who is doing 100+ runs although it is the tester's responsibility to follow the steps correctly. 

To address your points:
1. 
> Confusing ( Which one is being used, what if they are different)

and 

> Duplication ( Requires more maintenance and is more error prone. You make a change in one file and forget to push it to the other file.)

- I understand there being two file seems confusing and more maintenance. To remedy this, I suggest we actually remove the original file so there is only 1 file. The CDASim is currently REQUIRED to start from specific folder, so there is no breaking change if we have it in the folder and not in the image.

2. 
> Breaking Convention ( Every CARMA Config provides a VehicleConfigParam.yaml file that is built into the image volume, doing something different here may encourage other down the line to also diverge from this convention)

We are doing something new by testing things in simulation now. Just because it is convention doesn't mean it's perfect. We already broke the convention by having all these parameters in one place starting from k900, and it actually served us well. The need for such changes indicate there is something lacking from our current configuration management. It may NOT be the perfect solution that will stick, but I see it as a low-risk experimentation towards that goal.

**Review by MishkaMN — CHANGES_REQUESTED** [review]
My reasoning to have it in the config folder is because using carma-config image itself has drawbacks.

    Just changing one parameter is not user friendly at all -> change file + build image + set image just to exactly mirror what I have locally. I'd say let's just change the file and be done with it.
    More importantly, I imagine it is also error prone because the tester may forget to do those extra steps because all other parameter changes only require local file changes. So they may end up doing many runs until after the fact. This error is hard to notice because the actual value being used in the system is hidden from the user although local file shows updated value. We have an opportunity to lessen that room for error for the tester who is doing 100+ runs although it is the tester's responsibility to follow the steps correctly.

To address your points:
1.

    Confusing ( Which one is being used, what if they are different)

and

    Duplication ( Requires more maintenance and is more error prone. You make a change in one file and forget to push it to the other file.)

    I understand there being two file seems confusing and more maintenance. To remedy this, I suggest we actually remove the original file so there is only 1 file. The CDASim is currently REQUIRED to start from specific folder, so there is no breaking change if we have it in the folder and not in the image.

    Breaking Convention ( Every CARMA Config provides a VehicleConfigParam.yaml file that is built into the image volume, doing something different here may encourage other down the line to also diverge from this convention)

We are doing something new by testing things in simulation now. Just because it is convention doesn't mean it's perfect. We already broke the convention by having all these parameters in one place starting from k900, and it actually served us well. The need for such changes indicate there is something lacking from our current configuration management. It may NOT be the perfect solution that will stick, but I see it as a low-risk experimentation towards that goal.

**Comment by paulbourelly999** [comment]
> My reasoning to have it in the config folder is because using carma-config image itself has drawbacks.
> 
> * Just changing one parameter is not user friendly at all -> change file + build image + set image just to exactly mirror what I have locally. I'd say let's just change the file and be done with it.
> * More importantly, I imagine it is also error prone because the tester may forget to do those extra steps because all other parameter changes only require local file changes. So they may end up doing many runs until after the fact. This error is hard to notice because the actual value being used in the system is hidden from the user although local file shows updated value. We have an opportunity to lessen that room for error for the tester who is doing 100+ runs although it is the tester's responsibility to follow the steps correctly.
> 
> To address your points: 1.
> 
> > Confusing ( Which one is being used, what if they are different)
> 
> and
> 
> > Duplication ( Requires more maintenance and is more error prone. You make a change in one file and forget to push it to the other file.)
> 
> * I understand there being two file seems confusing and more maintenance. To remedy this, I suggest we actually remove the original file so there is only 1 file. The CDASim is currently REQUIRED to start from specific folder, so there is no breaking change if we have it in the folder and not in the image.
> 
> 
> > Breaking Convention ( Every CARMA Config provides a VehicleConfigParam.yaml file that is built into the image volume, doing something different here may encourage other down the line to also diverge from this convention)
> 
> We are doing something new by testing things in simulation now. Just because it is convention doesn't mean it's perfect. We already broke the convention by having all these parameters in one place starting from k900, and it actually served us well. The need for such changes indicate there is something lacking from our current configuration management. It may NOT be the perfect solution that will stick, but I see it as a low-risk experimentation towards that goal.

This is not new. The correct way to change carma configuration is to use `carma config edit` which does not require a rebuild of carma-config. Additional,  simulation being a deviation from traditional carma deployment does not grant us a blank check to break all convention. For other services we did not break convention because there is no convention. CARMA Config as up until now only handled vehicle configurations and does not have a convention for handling any other configuration file. 

If a tester wants to add custom volumes to their deployment for testing I think that is perfectly fine but to commit this as a released changes is very high risk. All follow on simulation scenarios will build on our configuration setup and probably end up slightly diverging from that bade on need and developer preference. Having this in here will contaminate every downstream simulation configuration image with the same duplication

**Review by MishkaMN — APPROVED** [review]
Okay I understand we don't want to break any convention here and give ample time for any new configuration management to be tested and put into practice before actually releasing them. The configs here looks good to me.

**Pull request merged**